python/version_details/shotgun_entities/card_widget.py

```diff
@@ -186,20 +187,27 @@ def remove_field(self, field_name):
 
         field_widget.hide()
         self.ui.field_grid_layout.removeWidget(field_widget)
+        field_widget.setParent(None)
+        field_widget.deleteLater()
 
         # If there's a label, then also remove that.
         field_label = self._fields[field_name]["label"]
 
         if field_label:
             field_label.hide()
             self.ui.field_grid_layout.removeWidget(field_label)
+            field_label.setParent(None)
+            field_label.deleteLater()
```

Comment:
I suspect this pattern is to plug a memory leak. I don't have first-hand experience with it, but I think it's reasonable to use it here given the situation.